astropy/utils/data_info.py

```diff
@@ -31,8 +31,8 @@
 from ..extern.six.moves import zip, cStringIO as StringIO
 
 # Tuple of filterwarnings kwargs to ignore when calling info
-IGNORE_WARNINGS = (dict(category=RuntimeWarning,
-                        module=r'numpy\.lib\.nanfunctions'),)
+IGNORE_WARNINGS = (dict(category=RuntimeWarning, message='All-NaN|'
+                        'Mean of empty slice|Degrees of freedom <= 0'),)
```

Comment:
I agree, but felt I should not change @taldcroft's code beyond what was strictly needed. (Note though that it still would need a tuple around it as well, as this is expected to be a tuple of `dict`).